src/components/Search/StatusFilter.tsx

```diff
@@ -2,39 +2,48 @@ import React, { useMemo } from 'react'
 
 import { useLocation } from '@gatsbyjs/reach-router'
 import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import isEqual from 'lodash/isEqual'
+import toSnakeCase from 'lodash/snakeCase'
 
-import { ProposalStatus, toProposalStatus } from '../../entities/Proposal/types'
-import locations from '../../modules/locations'
+import { GrantStatus } from '../../entities/Grant/types'
+import { ProposalStatus } from '../../entities/Proposal/types'
 
 import { FilterProps } from './CategoryFilter'
 import CollapsibleFilter from './CollapsibleFilter'
 import FilterLabel from './FilterLabel'
 
-export default React.memo(function StatusFilter({ onChange }: FilterProps) {
+type StatusType = typeof ProposalStatus | typeof GrantStatus
+
+export default React.memo(function StatusFilter({
+  onChange,
+  startOpen,
+  statusType,
+}: FilterProps & { statusType: StatusType }) {
   const t = useFormatMessage()
   const location = useLocation()
   const params = useMemo(() => new URLSearchParams(location.search), [location.search])
-  const status = toProposalStatus(params.get('status')) ?? undefined
+  const status = params.get('status')
 
-  function handleStatusFilter(status: ProposalStatus | null) {
+  function handleStatusFilter(status: string | null) {
```

Comment:
This is really similar to

```typescript
function handleTypeFilter(type: string | null, params: URLSearchParams) {
  const newParams = new URLSearchParams(params)
  type ? newParams.set('type', type) : newParams.delete('type')
  newParams.delete('page')
  const stringParams = newParams.toString()
  return `${location.pathname}${stringParams === '' ? '' : '?' + stringParams}`
}
```

I think the only difference is the "filter key". 
Could you extract it to a common fn? 🙏 
I think we could have a file for common functions to filters.

You are also using `Object.values` below, and there's this function defined in the other filter, that could be reused:

```typescript
function getValues<T extends object>(e: T): Array<T[keyof T]> {
  return Object.values(e)
}
```